### PR TITLE
[5.0] Remove 'track_errors' and $php_errormsg

### PR DESCRIPTION
**Pull request by Denitz**
Pull Request for Issue #41889 #25789 #17856 #33185

### Summary of Changes

Remove suppressing of methods which produce E_WARNING and necessity of catching the error in deprecated `$php_errormsg`.

I didn't update `Joomla\CMS\Filesystem\Stream` because it's deprecated since 4.4

### Testing Instructions

Try to parse language string with errors (with enabled Joomla debug + language debug)
Try to install the package from invalid URL

### Actual result BEFORE applying this Pull Request

See native Joomla error messages in installer and debug errors in language debug bar.

### Expected result AFTER applying this Pull Request

See the same errors/messages.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
I'm not sure if it's the right approach but it looks ok to me. @laoneo @wilsonge @Fedik opinions on this?

File: libraries/src/Language/Language.php [hunk 1]
```diff
@@ -699,19 +699,26 @@ protected function loadLanguage($fileName, $extension = 'unknown')
     /**
      * Parses a language file.
      *
-     * @param   string  $fileName  The name of the file.
```
Comment (HLeithner) [inline libraries/src/Language/Language.php]: please keep this var camel case

File: libraries/src/Language/Language.php [hunk 2]
```diff
@@ -755,7 +759,7 @@ public function debugFile(string $filename): int
             $line = trim($line);
 
             // Ignore comment lines.
-            if (!\strlen($line) || $line['0'] == ';') {
```
Comment (HLeithner) [inline libraries/src/Language/Language.php]: please don't do unrelated changes

File: libraries/src/Language/Language.php (hunk 1 above)
Comment (Denitz) [reply libraries/src/Language/Language.php]: `class Language extends BaseLanguage`
`use Joomla\Language\Language as BaseLanguage;`

The signature of parent's `Joomla\Language\Language` is  `parse($filename)`, it's not a good idea to change signature in this way, even though it's possible. Anyway, reverted.

File: libraries/src/Language/Language.php (hunk 2 above)
Comment (Denitz) [reply libraries/src/Language/Language.php]: Reverted.

**Comment by Denitz** [comment]
@HLeithner All done.

**Comment by HLeithner** [comment]
thanks I merged this for now so we have it in beta3. CI test should at least check the language parsing part.

thanks
